src/System.Reflection.DispatchProxy/src/System/Reflection/DispatchProxyGenerator.cs

```diff
@@ -947,16 +947,33 @@ public sealed override bool Equals(MethodInfo left, MethodInfo right)
 
                     // This assembly should work in netstandard1.3,
                     // so we cannot use MemberInfo.MetadataToken here.
-                    // Therefore, it compares honestly.
+                    // Therefore, it compares honestly referring ECMA-335 I.8.6.1.6 Signature Matching.
                     if (!Equals(left.DeclaringType, right.DeclaringType))
                         return false;
 
                     if (!Equals(left.ReturnType, right.ReturnType))
                         return false;
 
-                    if (left.Name != right.Name)
+                    if (left.CallingConvention != right.CallingConvention)
                         return false;
 
+                    if (left.IsStatic != right.IsStatic)
+                        return false;
+
+                    if ( left.Name != right.Name)
+                        return false;
+
+                    Type[] leftGenericArguments = left.GetGenericArguments();
```

Comment:
Nit: Since we're comparing method definitions, not instantiations, "leftGenericParameters" would be a more apt name.